### ISSUE TITLE
Enhancement: [no-useless-template-literals] Add fixing suggestions

### ISSUE THREAD

**Issue by StyleShit** [typescript-eslint/typescript-eslint#8056]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-useless-template-literals

### Description

Add suggestions to auto-fix the invalid cases

Splits from #7957

### Fail

```TypeScript
const a = `${ 1 }`;
const b = `${ null }`;
const c = `a-${ true }`;
```

### Pass

```TypeScript
const a = `1`;
const b = `null`;
const c = `a-true`;
```
